### PR TITLE
Use jUnit5 based HiveMetastoreExtension with HiveTests, remove HiveMetastoeTest

### PR DESCRIPTION
**Pull request by nk1506**

### CODE REVIEW

File: hive-metastore/src/test/java/org/apache/iceberg/hive/HiveCreateReplaceTableTest.java [hunk 1]
```diff
@@ -60,8 +66,23 @@ public class HiveCreateReplaceTableTest extends HiveMetastoreTest {
 
   private String tableLocation;
 
+  @RegisterExtension
+  public static final HiveMetastoreExtension HIVE_METASTORE_EXTENSION =
```
Comment (nastra) [inline hive-metastore/src/test/java/org/apache/iceberg/hive/HiveCreateReplaceTableTest.java]: this should be private

File: hive-metastore/src/test/java/org/apache/iceberg/hive/HiveCreateReplaceTableTest.java [hunk 2]
```diff
@@ -60,8 +66,23 @@ public class HiveCreateReplaceTableTest extends HiveMetastoreTest {
...
+      new HiveMetastoreExtension(DB_NAME, Collections.emptyMap());
+
+  protected HiveCatalog catalog;
```
Comment (nastra) [inline hive-metastore/src/test/java/org/apache/iceberg/hive/HiveCreateReplaceTableTest.java]: should be private

File: hive-metastore/src/test/java/org/apache/iceberg/hive/HiveCreateReplaceTableTest.java (hunk 1 above)
Comment (nk1506) [reply hive-metastore/src/test/java/org/apache/iceberg/hive/HiveCreateReplaceTableTest.java]: I made public as per [reference](https://junit.org/junit5/docs/5.1.1/api/org/junit/jupiter/api/extension/RegisterExtension.html)

Comment (nastra) [reply hive-metastore/src/test/java/org/apache/iceberg/hive/HiveCreateReplaceTableTest.java]: The reference you pointed to defines the `RegisterExtension` interface via `public @interface RegisterExtension` but that doesn't mean we need to make it public here where it is used. That being said, it should be defined `private` here and in all other places (similar to how it's defined in the `TestHiveCatalog` from your other PR

File: hive-metastore/src/test/java/org/apache/iceberg/hive/HiveCreateReplaceTableTest.java (hunk 2 above)
Comment (nastra) [reply hive-metastore/src/test/java/org/apache/iceberg/hive/HiveCreateReplaceTableTest.java]: please also align all other tests the same way

**Pull request closed without merging**